packages/contracts/contracts/L1/Lib_WithdrawalVerifier.sol

```diff
@@ -0,0 +1,57 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import { L2OutputOracle } from "./L2OutputOracle.sol";
+import {
+    Lib_SecureMerkleTrie
+} from "../../lib/optimism/packages/contracts/contracts/libraries/trie/Lib_SecureMerkleTrie.sol";
+
+/**
+ * @title WithdrawalVerifier
+ */
+library WithdrawalVerifier {
+    struct OutputRootProof {
+        uint256 timestamp;
+        bytes32 version;
+        bytes32 stateRoot;
+        bytes32 withdrawerStorageRoot;
+        bytes32 latestBlockhash;
+    }
+
+    function _verifyWithdrawerStorageRoot(
+        bytes32 _outputRoot,
+        OutputRootProof calldata _outputRootProof
+    ) internal pure returns (bool) {
+        return
+            _outputRoot ==
+            keccak256(
+                abi.encode(
+                    _outputRootProof.version,
+                    _outputRootProof.stateRoot,
+                    _outputRootProof.withdrawerStorageRoot,
+                    _outputRootProof.latestBlockhash
+                )
+            );
+    }
+
+    function _verifyWithdrawalInclusion(
+        bytes32 _withdrawalHash,
+        bytes32 _withdrawerStorageRoot,
+        bytes calldata _withdrawalProof
+    ) internal pure returns (bool) {
+        bytes32 storageKey = keccak256(
+            abi.encode(
+                _withdrawalHash,
+                uint256(1) // The withdrawals mapping is at the second slot in the layout
+            )
+        );
```

Comment:
Yes, though to be precise `M` there is just the ordering of all global storage variables (whether mapping or otherwise). So since the withdrawer has the [`nonce`](https://github.com/ethereum-optimism/optimistic-specs/blob/6839d522620483f10baf5427e055ac0627ced68d/packages/contracts/contracts/L2/Withdrawer.sol#L16) in the `0`th slot, this is at slot `1`. 
[Ref](https://docs.soliditylang.org/en/v0.8.4/internals/layout_in_storage.html#mappings-and-dynamic-arrays).